src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1123,6 +1123,10 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                     scriptConfig.Functions.Add((string)function);
                 }
             }
+            else
+            {
+                scriptConfig.Functions = null;
```

Comment:
It's re-used in ScriptHostManager... these are e2e tests. I'd expect this to repro in production.